lib/3scale/backend/metric/collection.rb

```diff
@@ -40,19 +40,25 @@ def parse_usage(raw_usage)
           end
         end
 
+        # Propagates the usage to all the levels of the hierarchy.
+        # For example, in this scenario:
+        # m1 --child_of--> m2 --child_of--> m3
+        # If there's a +1 in m1, this method will set the +1 in the other 2 as
+        # well.
         def process_parents(usage)
-          usage.keys.inject(usage.dup) do |memo, id|
-            p_id = parent_id(id)
-            if p_id
-              if Usage.is_set? memo[id]
-                memo[p_id] = memo[id]
+          usage.inject(usage.dup) do |memo, (id, val)|
+            is_set_op = Usage.is_set?(val)
+
+            while (id = parent_id(id))
```

Comment:
parens not needed here

Reply:
https://github.com/rubocop-hq/ruby-style-guide#safe-assignment-in-condition